neuraxle/metaopt/random.py

```diff
@@ -377,3 +383,166 @@ def transform(self, data_inputs):
         if self.best_validation_wrapper_of_model is None:
             raise Exception('Cannot transform RandomSearch before fit')
         return self.best_validation_wrapper_of_model.wrapped.transform(data_inputs)
+
+
+class HyperparamsRepository(ABC):
+    """
+    Hyperparams repository that saves hyperparams, and scores for every AutoML trial.
+
+    .. seealso::
+        :class:`HyperparamsRepository`,
+        :class:`HyperparameterSamples`,
+        :class:`AutoMLSequentialWrapper`
+    """
+
+    @abstractmethod
+    def create_new_trial(self, hyperparams: HyperparameterSamples):
+        """
+        Create new hyperperams trial.
+
+        :param hyperparams: hyperparams
+        :type hyperparams: HyperparameterSamples
+        :return:
+        """
+        pass
+
+    @abstractmethod
+    def load_all_trials(self) -> Tuple[List[HyperparameterSamples], List[float]]:
+        """
+        Load all hyperparameter trials with their corresponding score.
+
+        :return: (hyperparams, scores)
+        """
+        pass
+
+    @abstractmethod
+    def save_score_for_trial(self, hyperparams: HyperparameterSamples, score: float):
```

Comment:
maybe also methods like save_fail_for_trial

Reply:
Note that in hyperopt, each trial has a status. Like "STARTED", "FAILED", and "SUCCESS", etc. Maybe also "PLANNED", not sure.